docs/02-app/02-api-reference/04-functions/redirect.mdx

```diff
@@ -36,7 +36,23 @@ The `type` parameter has no effect when used in Server Components.
 
 ## Returns
 
-`redirect` does not return a value.
+`redirect` does not return any value. Instead, it throws a `NEXT_REDIRECT` error. When invoking the `redirect()` function within a `try/catch` block, you must invoke it in the `finally` block to ensure proper execution.
```

Comment:
I wonder if we should just put it after the try catch versus finally (that's what the other code examples have done). That work for you?

Reply:
I want to emphasize that `redirect()` throws an error, and this error will be caught by the `catch` block. That's what didn't click for me at first, I didn't think a function named `redirect` would throw an error. My use case was:

```
"use server";
import { redirect } from "next/navigation";

export const redirectOnResponse = () => {
  try {
    const res = new Response(null, { status: 200 });
    if (!res.ok) {
      throw new Error();
    }
    redirect("/success");
  } catch (e) {
    console.error(e);
    redirect("/error");
  }
};
```

Meaning the redirect was dependent on `Response`, and in this example always redirected to `/error`.

Wdyt of:

`redirect` does not return any value. Instead, it throws a `NEXT_REDIRECT` error. When using the `redirect()` function within a `try/catch` block keep these points in mind:

* Don't place the `redirect()` function inside a `try` block if you want the redirection to occur.

* If you need to use `redirect()` after error handling, place it in a `finally` block.

* For straightforward redirects, you can use redirect() outside of any `try/catch` block.